metasmoke.py

```diff
@@ -32,16 +32,98 @@
 import metasmoke_cache
 
 
-MAX_FAILURES = 10  # Preservative, 10 errors = MS down
 NO_ACTIVITY_PINGS_TO_STANDBY = 8
 NO_ACTIVITY_PINGS_TO_REPORT = 4
 
 
 # noinspection PyClassHasNoInit,PyBroadException,PyUnresolvedReferences,PyProtectedMember
 class Metasmoke:
     status_pings_since_scan_activity = 0
-    last_ping_scan_time = 0
-    last_ping_posts_scanned = 0
+    scan_stat_snapshot = None
+
+    class AutoSwitch:
+        """ Automatically switch metasmoke status """
+        MAX_FAILURES = 10  # 10 failures == ms down
+        MAX_SUCCESSES = 1  # 1 success == ms up
+        ping_failure_counter = 0  # Negative values indicate consecutive successes
+        autoswitch_is_on = True
+        rw_lock = threading.Lock()
+
+        @staticmethod
+        def ping_failed():
+            """ Indicate a metasmoke status ping connection failure """
+            with Metasmoke.AutoSwitch.rw_lock:
+                if Metasmoke.AutoSwitch.ping_failure_counter < 0:
+                    # Consecutive counter. Switch sign.
+                    Metasmoke.AutoSwitch.ping_failure_counter = 0
+                Metasmoke.AutoSwitch.ping_failure_counter += 1
+                current_counter = Metasmoke.AutoSwitch.counter
```

Comment:
What's `Metasmoke.AutoSwitch.counter` here? It appears to only ever be used in line 107, where it's set to 0. It looks like this (and line 107) should be `Metasmoke.AutoSwitch.ping_failure_counter`.

Reply:
Yes, this is the bug! (we literally noticed that at the same time.)